### PR TITLE
Update to latest c2pa-rs version

### PR DESCRIPTION
**Pull request by tmathern**
- Update to latest c2pa-rs version
- Add one more test to verify custom model data_type extraction
- Fix two other failing tests (failing after the c2pa-rs version update) since it seems the tested value legitimately changed

I ran builds locally, seems it also updated most of the docs. Let me know if that should be avoided or is OK to happen on rebuild.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-    "declaration": true,      
+    "declaration": true,
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
     "lib": ["es2021", "ES2022.Error"],
     "module": "Node16",
-    "moduleResolution": "node",
-    "noImplicitAny": true,    
-    "outDir": "dist",         
+    "moduleResolution": "node16",
```
Comment (tmathern) [inline tsconfig.json:9]: Small build fix here

File: tests/fixtures/certs/es256.pub [hunk 2]
```diff
@@ -29,18 +29,3 @@ BBQOfI1yZr+iyeUAlO+1boDitg4vazAfBgNVHSMEGDAWgBRembiG4Xgb2VcVWnUA
 UrYpDsuojDAKBggqhkjOPQQDAgNJADBGAiEAtdZ3+05CzFo90fWeZ4woeJcNQC4B
 84Ill3YeZVvR8ZECIQDVRdha1xEDKuNTAManY0zthSosfXcvLnZui1A/y/DYeg==
 -----END CERTIFICATE-----
```
Comment (tmathern) [inline tests/fixtures/certs/es256.pub:31]: @dyro Why/How did that change? Is it because I ran a build locally?

Comment (dyro) [reply tests/fixtures/certs/es256.pub:31]: I don't know - that shouldn't change automatically. Maybe revert this change?

File: Cargo.toml [hunk 3]
```diff
@@ -14,24 +14,24 @@ crate-type = ["cdylib"]
 
 [dependencies]
 async-trait = "0.1.77"
-c2pa = { version = "0.31.2", features = [
-	"fetch_remote_manifests",
-	"file_io",
-	"xmp_write",
+c2pa = { git = "https://github.com/contentauth/c2pa-rs", branch = "dyro/CAI-6506", features = [
```
Comment (dyro) [inline Cargo.toml:17]: Let's hold off on releasing until I get this change in main. It's already been approved. Will let you know when I've created a release!

**Review by dyro — APPROVED** [review]
Looks really clean, @tmathern. Approved, but wait on the merge until: 
1. You've added back the public key
2. I've published a new version c2pa-rs.

File: Cargo.toml (hunk 3 above)
Comment (tmathern) [reply Cargo.toml:17]: Sounds good!

**Comment by tmathern** [comment]
@dyro The public key is added back.

File: Cargo.toml (hunk 3 above)
Comment (tmathern) [reply Cargo.toml]: As the version is released, I updated this to latest release.

**Pull request merged**